src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/TalismanListener.java

```diff
@@ -239,7 +239,15 @@ public void onEnchant(EnchantItemEvent e) {
             TalismanEnchantment enchantment = talisman.getRandomEnchantment(e.getItem(), enchantments.keySet());
 
             if (enchantment != null) {
-                enchantments.put(enchantment.getEnchantment(), enchantment.getLevel());
+                /*
+                 * Fix #2679
+                 * By default, Bukkit doesn't allow book enchant any type enchantment.
+                 */
+                if (e.getItem().getType() == Material.BOOK) {
+                    e.getItem().addUnsafeEnchantment(enchantment.getEnchantment(), enchantment.getLevel());
```

Comment:
Did you test this? This seems a bit strange to me.

Reply:
https://hub.spigotmc.org/javadocs/spigot/org/bukkit/event/enchantment/EnchantItemEvent.html#getEnchantsToAdd()

> Get map of enchantment (levels, keyed by type) to be added to item (modify map returned to change values). Note: Any enchantments not allowed for the item will be **ignored**

Reply:
Already tested add enchantment in unsafe way, it will override a exist enchantment.